src/API/Reports/Customers/Stats/DataStore.php

```diff
@@ -93,32 +95,23 @@ public function get_data( $query_args ) {
 
 			$selections       = $this->selected_columns( $query_args );
 			$sql_query_params = $this->get_sql_query_params( $query_args );
+			// Clear SQL clauses set for parent class queries that are different here.
+			$this->subquery->clear_sql_clause( 'select' );
+			$this->subquery->add_sql_clause( 'select', 'SUM( gross_total ) AS total_spend,' );
+			$this->subquery->add_sql_clause(
+				'select',
+				'CASE WHEN COUNT( order_id ) = 0 THEN NULL ELSE COUNT( order_id ) END AS orders_count,'
+			);
+			$this->subquery->add_sql_clause(
+				'select',
+				'CASE WHEN COUNT( order_id ) = 0 THEN NULL ELSE SUM( gross_total ) / COUNT( order_id ) END AS avg_order_value'
```

Comment:
This looks to be a modification of the query for `avg_order_value` - I assume an improvement.. but should we select `0` if there are no orders rather than `NULL`?

Reply:
> but should we select 0 if there are no orders rather than NULL?

You can change it and run the unit tests to see that some fail. The outer query does an `AVG()`. `NULL` is not counted where a `0` is so the average calculates to the wrong amount.

I made the change because it's technically possible for `COUNT( order_id )` to be zero which would result in a divide by zero error.